test/test_component.py

```diff
@@ -17,6 +18,38 @@
 
 class TestComponent(unittest.TestCase):
 
+    def test_filter_by_roles(self):
+        """test the filter by roles utility"""
+        doc = sbol3.Document()
+        sbol3.set_namespace('http://sbolstandard.org/testfiles')
+        # create and add 3 components, with 2 having common role of dna
+        comp_1 = sbol3.Component('component_1', sbol3.SBO_DNA, roles=[tyto.SBO.deoxyribonucleic_acid])
+        comp_2 = sbol3.Component('component_2', sbol3.SBO_DNA, roles=[tyto.SO.engineered_region])
+        comp_3 = sbol3.Component('component_3', sbol3.SBO_DNA, roles=[tyto.SO.engineered_region, tyto.SBO.deoxyribonucleic_acid])
+        doc.add(comp_1)
+        doc.add(comp_2)
+        doc.add(comp_3)
+        # only comp_1 and comp_3 must be returned by the function
+        matched_itr = filter_top_level(doc, by_roles, required_role=tyto.SBO.deoxyribonucleic_acid)
+        matched = [obj for obj in matched_itr]
```

Comment:
You can write this more simply as just `matched = list(filter_top_level... `
(and below as well)

Reply:
Fixed